parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -36,19 +36,37 @@ configureTransparentHugePage() {
 
 {{- if ShouldConfigSwapFile}}
 configureSwapFile() {
-    SWAP_SIZE_KB=$(expr {{GetSwapFileSizeMB}} \* 1000)
-    DISK_FREE_KB=$(df /dev/sdb1 | sed 1d | awk '{print $4}')
-    if [[ ${DISK_FREE_KB} -gt ${SWAP_SIZE_KB} ]]; then
-        SWAP_LOCATION=/mnt/swapfile
-        retrycmd_if_failure 24 5 25 fallocate -l ${SWAP_SIZE_KB}K ${SWAP_LOCATION} || exit $ERR_SWAP_CREAT_FAIL
-        chmod 600 ${SWAP_LOCATION}
-        retrycmd_if_failure 24 5 25 mkswap ${SWAP_LOCATION} || exit $ERR_SWAP_CREAT_FAIL
-        retrycmd_if_failure 24 5 25 swapon ${SWAP_LOCATION} || exit $ERR_SWAP_CREAT_FAIL
-        retrycmd_if_failure 24 5 25 swapon --show | grep ${SWAP_LOCATION} || exit $ERR_SWAP_CREAT_FAIL
-        echo "${SWAP_LOCATION} none swap sw 0 0" >> /etc/fstab
+    # https://learn.microsoft.com/en-us/troubleshoot/azure/virtual-machines/troubleshoot-device-names-problems#identify-disk-luns
+    swap_size_kb=$(expr {{GetSwapFileSizeMB}} \* 1000)
+    symlinks=$(ls -la /dev/disk/azure)
+
+    if [[ -L /dev/disk/azure/resource-part1 ]]; then
+        echo "Will use resource disk for swap file"
```

Comment:
this makes me realize....we should be validating this on RP side. 

we shouldn't allow you to make a swap file bigger than $OS_DISK_SIZE_GB - 30 (because we require 30 for the os disk) if it's on the OS disk, and shouldn't allow > temp disk size if it's on the temp disk (or max of the two really)

Reply:
not relevant here

Reply:
yep that's a very fair point, is the size of the temp disk something we always know? (not sure if it's set by the customer or something we take of for them)

doing this on the RP side would also allow customers to actually have the first idea as to what's going on when their nodepools cant be provisioned because of this lol